lib/lz4.c

```diff
@@ -2002,13 +2003,13 @@ LZ4_decompress_generic(
 
         /* Fast loop : decode sequences as long as output < oend-FASTLOOP_SAFE_DISTANCE */
         DEBUGLOG(6, "using fast decode loop");
+        token = *ip++;
+        nextLength = token >> ML_BITS;  /* literal length */
         while (1) {
             /* Main fastloop assertion: We can always wildcopy FASTLOOP_SAFE_DISTANCE */
             assert(oend - op >= FASTLOOP_SAFE_DISTANCE);
             assert(ip < iend);
```

Comment:
Need to change this assert to `ip <= iend` because we have already incremented ip for the token read at this point.